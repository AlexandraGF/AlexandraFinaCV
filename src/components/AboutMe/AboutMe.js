import React, { Component } from 'react';
import plusImage from '../../plus.png';
import minusImage from '../../minus.png';
import './AboutMe.scss';

class AboutMe extends Component {
    constructor(props) {
        super(props);
        this.state = {clicked: false};
    
        this.handleClick = this.handleClick.bind(this);
      }

      handleClick() {
        this.setState(state => ({
          clicked: !state.clicked
        }))
    };

    render() {
        const statement = "Junior Full-Stack Software Developer available immediately; I worked as a Junior Developer until October 2018. Completed the Makers Bootcamp in Web Development in March 2018 and with a first-class Master’s in Strategic Business Management, keen to solve real-world problems through software – with excellent communication, analytical and creative skills. I have a good working understanding of Javascript and its framework ReactJs, Ruby; testing frameworks (including Jest, RSpec and Jasmine) and practical experience of applying them in team software development projects.";
        return (
        <div>
            <div className="statementHeading" onClick={this.handleClick}>
                <div className="statementTitle">Personal Statement</div><div className="statementIconImage"> {!this.state.clicked ? (<img src={plusImage} />) : (<img src={minusImage} />)}</div></div>
                    {this.state.clicked ? (<div className="statementBody">{statement}</div>) : "" } 
        </div>
        );
    }
}

export default AboutMe;