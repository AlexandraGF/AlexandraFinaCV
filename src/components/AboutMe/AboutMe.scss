@import '../../variables.scss';

.statementHeading {
    font-family: 'Allura', cursive;
    font-size: 30px;
    margin: 7% 7% 0 11%;
    padding-left: 6%;
    box-shadow: $boxShadow;
    border-radius: $borderRadius;
    display: inline-block;
    width: 80%;
}

.statementTitle {
    width: 80%;
}

.statementIconImage {
    width: 8%;
}

.statementTitle, .statementIconImage {
    flex-wrap: wrap;
    display: inline-flex;
}

.statementHeading:hover {
    transform: $hoverSize;
    cursor: $cursorPointer;
}

.statementBody {
    box-shadow: 5px 10px 12px #a2c3c3;
    margin: 7% 4% 7% 13%;
    padding: 2% 2% 2% 2%;
    text-align: justify;
    font-family: 'K2D', sans-serif;
}
