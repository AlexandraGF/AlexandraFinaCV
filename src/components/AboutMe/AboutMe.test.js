import React from 'react';
import ReactDOM from 'react-dom';
import Enzyme, {shallow, mount} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';

Enzyme.configure({ adapter: new Adapter() });

import AboutMe from './AboutMe';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<AboutMe />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('renders text', () => {
    const wrapper = mount(<AboutMe />);
    expect(wrapper.text()).toContain("Personal Statement");
});
    
it('renders AboutMe Component', () => {
const tree = renderer.create(
<AboutMe />
).toJSON();

expect(tree).toMatchSnapshot();
})
