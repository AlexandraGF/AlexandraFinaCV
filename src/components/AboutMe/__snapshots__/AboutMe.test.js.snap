// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders AboutMe Component 1`] = `
<div>
  <div
    className="statementHeading"
    onClick={[Function]}
  >
    <div
      className="statementTitle"
    >
      Personal Statement
    </div>
    <div
      className="statementIconImage"
    >
       
      <img
        src="plus.png"
      />
    </div>
  </div>
  
</div>
`;
